File: data/greedy/task-scheduler.py
from collections import Counter

def leastInterval(tasks: list[str], n: int) -> int:
    counts = Counter(tasks)
    max_freq = max(counts.values())
    max_freq_count = sum(1 for v in counts.values() if v == max_freq)
    
    return max(len(tasks), (max_freq - 1) * (n + 1) + max_freq_count)

File: data/greedy/task-scheduler.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <algorithm>

using namespace std;

int leastInterval(vector<char>& tasks, int n) {
    unordered_map<char, int> counts;
    int maxFreq = 0;
    for (char task : tasks) {
        counts[task]++;
        maxFreq = max(maxFreq, counts[task]);
    }
    
    int maxFreqCount = 0;
    for (auto& pair : counts) {
        if (pair.second == maxFreq) maxFreqCount++;
    }
    
    int intervals = (maxFreq - 1) * (n + 1) + maxFreqCount;
    return max((int)tasks.size(), intervals);
}
